### PR TITLE
ebos: implement partial elimination of systematic mass defects

### PR DESCRIPTION
**Pull request by andlaus**
the idea is to compensate the residual of the final solution of a time step by means of an opposing source term in the next time step. While performance seems to be a few percent worse than the master version when using the same settings, the results often seem to be much better, so we can often get away with more relaxed tolerances.

This patch has been developed as a joint project with @totto82 and @osae.

(Note that `flow` is unaffected by this because drift compensation is an experimental feature and thus disabled within the production simulator.)

### CODE REVIEW

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -1443,6 +1480,38 @@ public:
...
+            // compute the weighted total drift rate
+            Scalar totalDriftRate = 0.0;
+            for (unsigned eqIdx = 0; eqIdx < numEq; ++ eqIdx)
+                totalDriftRate +=
+                    std::abs(dofDriftRate[eqIdx])*dt*model.eqWeight(globalDofIdx, eqIdx)/poro;
+
+            // make sure that we do not exceed the maximum rate of drift compensation
+            if (totalDriftRate > maxCompensation)
+                dofDriftRate *= maxCompensation/totalDriftRate;
+
```
Comment (totto82) [inline ebos/eclproblem.hh]: The implementation here is slightly different from the earlier prototype.  How well did you test this? How sensitive is the results on the particular choice of maxCompensation?

Comment (andlaus) [reply ebos/eclproblem.hh]: I ran it on Norne and M2R5, and it seemed to work well. more extensive testing is definitely in order, though...

**Comment by andlaus** [comment]
jenkins build this please

File: ebos/eclproblem.hh (hunk 1 above)
Comment (totto82) [reply ebos/eclproblem.hh]: Ok. I will do some more testing. Since it is within the experimental mode I will merge it as is.

**Pull request merged**